util/src/org/commcare/util/cli/EntityScreen.java

```diff
@@ -113,10 +114,17 @@ public boolean setCurrentScreenToDetail() {
         return true;
     }
 
-    public void setCurrentScreenToDetail(int index) {
+    public void setCurrentScreenToDetail(int index) throws CommCareSessionException {
         EvaluationContext subContext = new EvaluationContext(mSession.getEvaluationContext(), this.mCurrentSelection);
 
-        this.mCurrentScreen = new EntityDetailSubscreen(index, this.mLongDetailList[index], subContext, getDetailListTitles(subContext));
+        TreeReference detailNodeset = this.mLongDetailList[index].getNodeset();
+        if (detailNodeset != null) {
+            TreeReference contextualizedNodeset = detailNodeset.contextualize(this.mCurrentSelection);
+            this.mCurrentScreen = new EntityListSubscreen(this.mLongDetailList[index], subContext.expandReference(contextualizedNodeset), subContext);
```

Comment:
This screws up the terminal outputs from this, right? IE: after choosing the list detail things go badly, and it shows an odd list of items that have 11) before and such?

Not sure how insistent to be on making the CLI functional for new features, but depending on LOE can you let me know whether you're planning on making it work more similarly to the existing details?